src/Microsoft.Azure.WebJobs.Host/Executors/FunctionExecutor.cs

```diff
@@ -125,15 +125,15 @@ public async Task<IDelayedException> TryExecuteAsync(IFunctionInstance functionI
                     logCompletedCancellationToken = cancellationToken;
                 }
 
-                if (instanceLogEntry != null)
+                if (functionCompletedMessage != null)
                 {
-                    await NotifyCompleteAsync(instanceLogEntry, functionCompletedMessage.Arguments, exceptionInfo);
-                    _resultsLogger?.LogFunctionResult(instanceLogEntry);
+                    await _functionInstanceLogger.LogFunctionCompletedAsync(functionCompletedMessage, logCompletedCancellationToken);
                 }
 
-                if (functionCompletedMessage != null)
+                if (instanceLogEntry != null)
                 {
-                    await _functionInstanceLogger.LogFunctionCompletedAsync(functionCompletedMessage, logCompletedCancellationToken);
+                    await NotifyCompleteAsync(instanceLogEntry, functionCompletedMessage.Arguments, exceptionInfo);
+                    _resultsLogger?.LogFunctionResult(instanceLogEntry);
                 }
```

Comment:
The change swaps these two lines -- so we call `LogFunctionCompletedAsync` before we call `LogFunctionResult`, which marks the function as completed in App Insights.